src/test/kotlin/org/wycliffeassociates/otter/common/domain/md/ParseMdTest.kt

```diff
@@ -0,0 +1,175 @@
+                            "Body 2"
+                    ),
+                    "expected" to arrayListOf(
+                            HelpResource("Title 1", "Body 1"),
+                            HelpResource("Title 2", "Body 2")
+                    )
+            ),
+            mapOf(
+                    "lines" to listOf(
+                            "# Title 1",
+                            "",
+                            "Body 1",
+                            "",
+                            "Body 2", // Second line of body text
+                            "",
+                            "# Title 3",
+                            "Body 3", // No space before body text
+                            "",
+                            "# Title 4", // Heading with no body text
+                            "",
+                            "# Title 5",
+                            "",
+                            "Body 5",
+                            "# Title 6", // No space before title text
+                            "",
+                            "Body 6"
+                    ),
+                    "expected" to arrayListOf(
+                            HelpResource("Title 1", "Body 1 Body 2"),
+                            HelpResource("Title 3", "Body 3"),
+                            HelpResource("Title 4", ""),
+                            HelpResource("Title 5", "Body 5"),
+                            HelpResource("Title 6", "Body 6")
+                    )
+            )
+    )
+
+    // Testing title text extraction
+    private val testGetTitleTextCases = listOf(
+            mapOf(
+                    "line" to "# Hello",
+                    "expected" to "Hello"
+            ),
+            mapOf(
+                    "line" to "#  Matthew",
+                    "expected" to "Matthew"
+            ),
+            mapOf(
+                    "line" to "## John said",
+                    "expected" to "John said"
+            ),
+            mapOf(
+                    "line" to "# John said # hello",
+                    "expected" to "John said # hello"
+            ),
+            mapOf(
+                    "line" to "#John said hello",
+                    "expected" to "John said hello"
+            )
+    )
+
+    // Testing title recognition
+    private val testIsTitleLineCases = listOf(
+            mapOf(
+                    "line" to "# Matthew",
+                    "expected" to true
+            ),
+            mapOf(
+                    "line" to "## Matthew",
+                    "expected" to true
+            ),
+            mapOf(
+                    "line" to "",
+                    "expected" to false
+            ),
+            mapOf(
+                    "line" to "Matthew # said hello",
+                    "expected" to false
+            ),
+            mapOf(
+                    "line" to "# ",
+                    "expected" to false
+            ),
+            mapOf(
```

Comment:
You're using `Map<String, String>` as an ad-hoc way to make dynamically typed objects. It works here since this is a small source file, but you lose type safety and null safety. All the `as String`s in the code below are a sign that you're working against Kotlin's null safety.

As an example, if you mistype "expetced" as your key, you'll get back a null String, which you then cast to String.  Like I said, it really isn't causing a problem because this is such a small file and any bug here is just going to make a test case fail.  NBD, and I know I'm being picky, but the first PR is a good time to talk about this stuff. 

Please experiment with converting all your `Map`s to data classes with `input` and `expected` fields.  You can make a single generic data class that handles either String or Bool types for `expected`, or you can just make two classes.

Or, perhaps better and simpler, since all of your classes really only have those two fields, you could just use [`to`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/to.html) to create input-to-expected pairs.

Reply:
Simply using to is a much better approach, thanks for the suggestion!